operator/controllers/btpoperator_controller.go

```diff
@@ -978,38 +1122,67 @@ func (r *BtpOperatorReconciler) deleteAllOfinstalledResources(ctx context.Contex
 	return nil
 }
 
-func (r *BtpOperatorReconciler) HandleReadyState(ctx context.Context, cr *v1alpha1.BtpOperator) error {
+func (r *BtpOperatorReconciler) prepareInstallInfo(ctx context.Context, cr *v1alpha1.BtpOperator) (types.InstallInfo, Reason, error) {
```

Comment:
you can update the CR status in the method if required and omit a Reason from returned values